src/Npgsql/Replication/ReplicationConnection.cs

```diff
@@ -360,80 +361,36 @@ async Task<TimelineHistoryFile> TimelineHistoryInternal(uint tli, CancellationTo
-            await buf.EnsureAsync(len);
-            var slotNameResult = buf.ReadString(len);
-
-            // consistent_point
-            await buf.EnsureAsync(4);
-            len = buf.ReadInt32();
-            Debug.Assert(len > 0, "consistent_point should never be empty");
-            await buf.EnsureAsync(len);
-            var consistentPoint   = NpgsqlLogSequenceNumber.Parse(buf.ReadString(len));
-
-            // snapshot_name
-            await buf.EnsureAsync(4);
-            len = buf.ReadInt32();
-            string? snapshotName;
-            if (len == -1)
-                snapshotName = null;
-            else
-            {
-                await buf.EnsureAsync(len);
-                snapshotName = buf.ReadString(len);
-            }
-
-            // output_plugin
-            await buf.EnsureAsync(4);
-            len = buf.ReadInt32();
-            if (len != -1)
-            {
-                await buf.EnsureAsync(len);
-                buf.Skip(len); // We know already what we created
-            }
-
-            Expect<CommandCompleteMessage>(await Connector.ReadMessage(true), Connector);
-            Expect<ReadyForQueryMessage>(await Connector.ReadMessage(true), Connector);
-
-            return new ReplicationSlotOptions(slotNameResult, consistentPoint, snapshotName);
+            var result = await ReadSingleRow(command, cancellationToken);
+            var slotName = (string)result[0];
+            var consistentPoint = (string)result[1];
+            var snapshotName = (string?)result[2];
+            return new ReplicationSlotOptions(slotName, NpgsqlLogSequenceNumber.Parse(consistentPoint), snapshotName);
         }
         catch (PostgresException e)
         {
-            if (!Connector.IsBroken && PostgreSqlVersion < FirstVersionWithTemporarySlotsAndSlotSnapshotInitMode && e.SqlState == PostgresErrorCodes.SyntaxError)
+            if (Connector.IsBroken || e.SqlState != PostgresErrorCodes.SyntaxError)
+                throw;
+
+            if ( PostgreSqlVersion < FirstVersionWithTwoPhaseSupport && command.Contains("TWO_PHASE"))
+                throw new NotSupportedException("Logical replication support for prepared transactions was introduced in PostgreSQL " +
+                                                FirstVersionWithTwoPhaseSupport.ToString(1) +
+                                                ". Using PostgreSQL version " +
+                                                (PostgreSqlVersion.Build == -1
+                                                    ? PostgreSqlVersion.ToString(2)
+                                                    : PostgreSqlVersion.ToString(3)) +
+                                                " you have to set the twoPhase argument to false.", e);
+            if (PostgreSqlVersion < FirstVersionWithTemporarySlotsAndSlotSnapshotInitMode)
             {
-                if (temporarySlot)
+                if (command.Contains("TEMPORARY"))
```

Comment:
Is this safe? Could the word TEMPORARY not appear e.g. in the slot name, or elsewhere?

I realize we were already doing this (e.g. with _SNAPSHOT), but it could be problematic...

Reply:
Good catch.
I'd say it's "relatively" safe because the worst that could happen would be the wrong exception if someone uses weird names for replication slots but still.
It's actually easy to make it safer by including one leading whitespace in all three cases because a slot name cannot contain whitespace.

Reply:
I'm not a huge fan of string searching hacks :) Especially since we have all the information we need...

How about just doing these checks can be done up-front (without waiting for PG to throw)? We could just have a function called ValidateSlotSettings, which accepts the various options as flags and checks if the PG version supports then?

Reply:
> I'm not a huge fan of string searching hacks

Neither am I.
The way I just pushed it is absolutely safe though.
I just found out that it even was safe before because replication slot names can only contain lower-case letters, numbers, and the underscore character.
See: https://www.postgresql.org/docs/current/warm-standby.html#STREAMING-REPLICATION-SLOTS-MANIPULATION

Reply:
Up to you then...